src/main/java/com/employeesService/EmployeesService/service/EmployeeServiceImpl.java

```diff
@@ -1,28 +1,56 @@
 package com.employeesService.EmployeesService.service;
 
+import com.employeesService.EmployeesService.exception.DepartmentNotFoundException;
 import com.employeesService.EmployeesService.exception.EmployeeNotFoundException;
+import com.employeesService.EmployeesService.model.Department;
+import com.employeesService.EmployeesService.model.DepartmentEmployee;
+import com.employeesService.EmployeesService.model.DepartmentEmployeeId;
 import com.employeesService.EmployeesService.model.Employee;
+import com.employeesService.EmployeesService.repository.DepartmentRepository;
 import com.employeesService.EmployeesService.repository.EmployeeRepository;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 
 import java.util.List;
 
 @Service
 public class EmployeeServiceImpl implements EmployeeService {
 
     private final EmployeeRepository employeeRepository;
+    private final DepartmentRepository departmentRepository;
 
     @Autowired
-    public EmployeeServiceImpl(EmployeeRepository employeeRepository) {
+    public EmployeeServiceImpl(EmployeeRepository employeeRepository, DepartmentRepository departmentRepository) {
         this.employeeRepository = employeeRepository;
+        this.departmentRepository = departmentRepository;
     }
 
     @Override
     public Employee save(Employee employee) {
         return employeeRepository.save(employee);
     }
 
+    @Override
+    @Transactional
+    public Employee addEmployeeToDepartment(String deptNo, Long empNo, DepartmentEmployee departmentEmployee) {
+        Department department = this.departmentRepository.findById(deptNo).orElseThrow(() -> new DepartmentNotFoundException(deptNo));
+        Employee employee = this.employeeRepository.findById(empNo).orElseThrow(() -> new EmployeeNotFoundException(empNo));
+        DepartmentEmployeeId id = new DepartmentEmployeeId(deptNo, empNo);
+        departmentEmployee.setId(id);
+        departmentEmployee.setDepartment(department);
+        departmentEmployee.setEmployee(employee);
+        department.addEmployee(departmentEmployee);
+        this.departmentRepository.save(department);
+        return employee;
+    }
```

Comment:
```suggestion
    @Override
    @Transactional
    public Employee addEmployeeToDepartment(String deptNo, Long empNo, DepartmentEmployee departmentEmployee) {
        Department department = this.departmentRepository.findById(deptNo).orElseThrow(() -> new DepartmentNotFoundException(deptNo));
        Employee employee = this.employeeRepository.findById(empNo).orElseThrow(() -> new EmployeeNotFoundException(empNo));
        department.addEmployee(employee);
        this.departmentRepository.save(department);
        return employee;
    }
```

Reply:
The following code should be inside the Department entity `addEmployee(employee, fromDate)` method

```
        DepartmentEmployeeId id = new DepartmentEmployeeId(deptNo, empNo);
        departmentEmployee.setId(id);
        departmentEmployee.setDepartment(department);
        departmentEmployee.setEmployee(employee);